frontend/src/app/core/services/forms/forms.service.ts

```diff
@@ -83,15 +83,29 @@ export class FormsService {
       .keys(resources)
       .reduce((result, resourceKey) => {
         const resource = resources[resourceKey];
+        let resourceValue;
         // Form.payload resources have a HalLinkSource interface while
         // API resource options have a IAllowedValue interface
-        const resourceValue = Array.isArray(resource) ?
-          resource.map(resourceElement => ({ href: resourceElement?.href || resourceElement?._links?.self?.href })) :
-          { href: resource?.href || resource?._links?.self?.href };
+        if (Array.isArray(resource)) {
+          const formattedResourceValue = resource.reduce((result:any[], resourceElement) => {
+            const href = resourceElement?.href || resourceElement?._links?.self?.href;
+            if (href) {
+              result = [...result, {href}]
```

Comment:
It'd be cooler if you return this immediately instead of modifying the argument array

Reply:
Especially because it shadows `result` in the outerlying reduce that starts on line 84

Reply:
Thanks, changed 👍